src/sdp.jl

```diff
@@ -90,8 +82,8 @@ julia> enclose_SumOfSquares(f, dom, order, backend=backend, QUIET=true)
 
 To get the runtime, use `MOI.get(model, MOI.SolveTime())`.
 """
-function enclose_SumOfSquares(f::Function, dom::Interval_or_IntervalBox;
-                              order::Int=5, backend=SDPA.Optimizer, kwargs...)
+function enclose_SumOfSquares(f::Function, dom::Interval_or_IntervalBox, backend;
+                              order::Int=5, kwargs...)
```

Comment:
Can we keep it a named keyword? (so: after the `;` but without a default value)

Reply:
I suggest to keep make it a named keyword for the API function (`enclose`), but keep it a positional argument for internal functions `_enclose` (here, `enclose_SumOfSquares`) and use dispatch. Basically, we shall change the current mechanism in the line of https://github.com/JuliaReach/RangeEnclosures.jl/issues/64  

I'm fine merging this PR and making the bigger design change in another one.

Reply:
yes. with #64 the main structure will change anyway, it will be sth like (abstract types left out for simplicity)
```julia
struct Solver1 end

struct Solver2 end

struct Solver3 end

enclose(f, X, solver1; kwargs_for_solver1)

enclose(f, X,  solver2; kwarfs_for_solver2)

enclose(f, X; kwargs....) = enclose(f, X, DefaultSolver; kwargs...)
```

alternatively, the parameters corresponding to each solver could be inside the struct and not have kwargs at all, but I guess this can be discussed further in the next PR?

keyword arguments without default value sounds a little strange, can you even do it? Alternatively one could give it `nothing` as default value adn throw an `ArgumentError("backend missing")` if it's nothing?

Reply:
> alternatively, the parameters corresponding to each solver could be inside the struct

yes, that's better imo

> keyword arguments without default value sounds a little strange, can you even do it?

i was thinking to use sth like `Base.@kwdef`

Reply:
> `enclose(f, X; kwargs....) = enclose(f, X, DefaultSolver; kwargs...)`

I thought that the whole point here is that there is no default solver anymore. How would that work?

> keyword arguments without default value sounds a little strange, can you even do it?

Yes, then they are required arguments (you have to specify them, but as a keyword). This reminds the user to specify a solver.

> Alternatively one could give it nothing as default value adn throw an ArgumentError("backend missing") if it's nothing?

Yes, that is also fine (because more informative).

>  alternatively, the parameters corresponding to each solver could be inside the struct

Yes, I also think that is the better (because more scalable) way.

Reply:
> I suggest to keep make it a named keyword for the API function (`enclose`), but keep it a positional argument for internal functions `_enclose` (here, `enclose_SumOfSquares`) and use dispatch.

`enclose_SumOfSquares` is documented and there is a `_enclose_SumOfSquares`, so I think this method is part of the public API. If we change it in #64, then never mind.

Reply:
> I thought that the whole point here is that there is no default solver anymore. How would that work?

sorry solver was a misleading word. Here I meant "default algorithm to compute the range enclosure", e.g. BranchAndPrune or IntervalOptimisation etc. If you choose to use the SumOfSquares approach, then you should also specify the backend, as discussed here

Reply:
@lucaferranti If you change back to a kwarg, I will approve.

Reply:
sorry, this fell off the radar yesterday, doing it now 🔨 🔧